### DISCUSSION TITLE
Frame pointers?

### DISCUSSION THREAD

**Discussion by aetimmes** [bottlerocket-os/bottlerocket#3841] [General]
Having recently read Brendan Gregg's [blog post] about frame pointers and seeing that Ubuntu and Fedora had both re-enabled them, I was curious - does Bottlerocket have frame pointers enabled currently? (I did some light `readelf` spelunking in an instance and I think the answer is "no", but I'm not 100% certain.) 

If not, what are the chances we could see them enabled in the near future? $DAY_JOB is moving workloads from Ubuntu to Bottlerocket and I'd love to maintain the integrity of our flamegraphs.

[blog post]: https://www.brendangregg.com/blog/2024-03-17/the-return-of-the-frame-pointers.html

**Comment by bcressey** [comment]
I'm open to it! I've been keeping an eye on this effort in Fedora and elsewhere and wouldn't mind seeing Bottlerocket join the club.

The change would need to happen over in the SDK, in [macros/shared](https://github.com/bcressey/bottlerocket-sdk/blob/36aa70c3966e8489a5a105a3953f994ac0caafdd/macros/shared#L8) where the common CFLAGS are set.

Plus one or two places where those CFLAGS aren't used; somewhat inevitably, glibc - maybe the most important one - has its own set of flags and would need a separate patch.

I'm not sure if that'd be sufficient on its own or if we also need to publish debug symbols somewhere. That's been on my wishlist since 2020 and is somewhat more complicated. (What I want to do is upload debug symbols to S3 in a debuginfod-compatible layout, stick a CloudFront distribution in front of it, and have that just work. But I haven't had a chance to prioritize this.)

But we can start with including frame pointers and see how far that gets us.
